dspace-api/src/main/java/org/dspace/discovery/SolrServiceSupervisionOrderIndexingPlugin.java

```diff
@@ -0,0 +1,64 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.discovery;
+
+import java.sql.SQLException;
+import java.util.List;
+import java.util.Objects;
+
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.solr.common.SolrInputDocument;
+import org.dspace.content.Item;
+import org.dspace.core.Context;
+import org.dspace.discovery.indexobject.IndexableInProgressSubmission;
+import org.dspace.discovery.indexobject.IndexableWorkflowItem;
+import org.dspace.discovery.indexobject.IndexableWorkspaceItem;
+import org.dspace.supervision.SupervisionOrder;
+import org.dspace.supervision.service.SupervisionOrderService;
+import org.springframework.beans.factory.annotation.Autowired;
+
+/**
+ *
+ * 
+ * @author Mohamed Eskander (mohamed.eskander at 4science dot it)
+ */
+public class SolrServiceSupervisionOrderIndexingPlugin implements SolrServiceIndexPlugin {
+
+    @Autowired(required = true)
+    private SupervisionOrderService supervisionOrderService;
+
+    @Override
+    public void additionalIndex(Context context, IndexableObject indexableObject, SolrInputDocument document) {
+        try {
+
+            if (!(indexableObject instanceof IndexableWorkspaceItem) &&
+                !(indexableObject instanceof IndexableWorkflowItem)) {
```

Comment:
I have reviewed our initial notes about that. The reason why we need to keep track of the supervision also when the item is in workflow is that workflow item can be rejected back to the workspaceitem. The idea is that if an item was supervised it should be still supervised when it goes back. In version 6 this was managed in an inconsistent way as the resource policies related to the supervision was left in place and so available when the item was sent back but the workspaceitem2eperson link was of course loss.
Please note that keep the information about the supervision when the item is in workflow add no functional changes as the supervisors cannot do anything to the workflowitem. This can be eventually proven by a dedicated test

Reply:
@abollini : That makes sense as a use case for why a Supervision Order might remain on a `WorkflowItem`.  I can agree to that.

In that case, I'd recommend we create more tests which align with this use case / prove that it works as expected.  For instance, based on this use case, I'd assume the following should be true:
1. When an Item moves from WorkspaceItem to WorkflowItem, the SupervisionOrder is retained.  However, the users/groups related to the SupervisionOrder only have `READ` privileges on the Item when it is in Workflow (in the same way that a Submitter can only "read" their submission once it is in Workflow & can no longer modify it unless it is rejected & returned to "WorkspaceItem" status).  
    * In other words, while in WorkspaceItem status, a SupervisionOrder can allow for edit privileges. But, once in WorkflowItem status, it only allows for read privileges.
2. Obviously, when the Item becomes archived, the SupervisionOrder should be deleted.

Reply:
Closing this request... I've verified these new tests have been added.